apps/VideoStream/videoStream.py

```diff
@@ -8,36 +8,32 @@ class Connections:
     """Class for handling active connections"""
 
     def __init__(self, timeout):
-        self.connections = []
-        self.times_since_heartbeat = []
+        self.connections = {}
         self.timeout = timeout
         self.lock = threading.Lock()
 
     def add(self, new_address):
-        self.connections.append(new_address)
-        self.times_since_heartbeat.append(time.time())
-        
+        key = new_address[0] + ':' + str(new_address[1])
+        connection = {'socket':new_address, 'time':time.time()}
+        self.connections[key] = connection
+
     def remove(self, to_remove):
-        if (to_remove is str):
-            to_remove = self.connections.index(to_remove)
+        # if (to_remove is str):
+        #     to_remove = self.connections.index(to_remove)
         del self.connections[to_remove]
-        del self.times_since_heartbeat[to_remove]
-
-    def read_heartbeat(self, address):
-        index = self.connections.index(address)
-        self.times_since_heartbeat[index] = time.time()
 
     def cleanup(self):
-        i = 0
-        while i < len(self.times_since_heartbeat):
-            if(time.time() - self.times_since_heartbeat[i] > self.timeout):
-                print("Client", self.connections[i], "timed out.")
-                self.remove(i)
-            i = i + 1
+        to_remove = []
```

Comment:
Had to make a list of the connections to be removed. Removing them during the for loop caused a runtime error due to the dictionary changing in size.